sdk/storage/azure-storage-common/inc/azure/storage/common/paged_response.hpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <memory>
+#include <string>
+#include <type_traits>
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/raw_response.hpp>
+
+namespace Azure { namespace Storage {
+
+  namespace _internal {
+    template <class Derived> class PageResult {
+    public:
+      std::unique_ptr<Azure::Core::Http::RawResponse> RawResponse;
+      std::string NextPageToken;
+
+      bool HasMore() const { return !NextPageToken.empty(); }
+
+      void NextPage()
```

Comment:
I'm thinking about adding an overload that accepts a `Context`, so that user can use a different context for each HTTP request